src/bin/tools/transformCodebase.ts

```diff
@@ -20,27 +20,29 @@ export function transformCodebase(params: { srcDirPath: string; destDirPath: str
         }))
     } = params;
 
-    for (const file_relative_path of crawl(srcDirPath)) {
+    const filePromises = (await crawl(srcDirPath)).map(async file_relative_path => {
```

Comment:
maybe you can use `walk.ts`?

`
await Promise.all(Array.from(walk(srcDirPath)).map(async relativeFilePath => {
 ....
})))
`

Reply:
```suggestion
    const filePromises = (await crawl(srcDirPath)).map(async fileRelativePath => {
```

Reply:
To save some real time I'd like an api more like 

```ts
export type Crawl = (onFileFound: (fileRelativePath: string)=> void)=> Promise<void>;
```

It would be faster than having to collect all file to start processing them.